domain/model/service/modelservice.go

```diff
@@ -46,23 +55,26 @@ type ControllerState interface {
 // ModelService defines a service for interacting with the underlying model
 // state, as opposed to the controller state.
 type ModelService struct {
-	clock        clock.Clock
-	modelID      coremodel.UUID
-	controllerSt ControllerState
-	modelSt      ModelState
+	clock                 clock.Clock
+	modelID               coremodel.UUID
+	controllerSt          ControllerState
+	modelSt               ModelState
+	environProviderGetter func(cloudType string) (environs.EnvironProvider, error)
```

Comment:
We should look to define the specific scope of what we are trying to test. Specifically `environs.EnvironProvider` is far too big and we don't need anything on this interface. In this package I would define an interface with the set of things we need and be asking for that.

You could if you wanted to also define a type for the func.